src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,78 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { Filters } from '../../types/Filters';
+
+type Props = {
+  onFilterChange: (newFilter: Filters) => void,
+  onQueryChange: (newQuery: string) => void,
+  query: string,
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  onFilterChange,
+  onQueryChange,
+  query,
+}) => {
+  const handleFilterChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    if (event.target.value === Filters.All
+      || event.target.value === Filters.Completed
+      || event.target.value === Filters.Active) {
+      onFilterChange(event.target.value);
+    }
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onQueryChange(event.target.value);
+  };
+
+  const handleQueryClear = () => {
+    onQueryChange('');
+  };
```

Comment:
from your names of the functions i don't get **at all** what do these functions do. + they are at the top level, so it takes a while for the reader to understand what's happening here. It shouldn't be like that